metricbeat/module/rabbitmq/exchange/exchange_test.go

```diff
@@ -0,0 +1,91 @@
+package exchange
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFetchEventContents(t *testing.T) {
+	absPath, err := filepath.Abs("../_meta/testdata/")
+
+	response, err := ioutil.ReadFile(absPath + "/exchange_sample_response.json")
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		w.Header().Set("Content-Type", "application/json;")
+		w.Write([]byte(response))
+	}))
+	defer server.Close()
+
+	config := map[string]interface{}{
+		"module":     "rabbitmq",
+		"metricsets": []string{"exchange"},
+		"hosts":      []string{server.URL},
+	}
+
+	f := mbtest.NewEventsFetcher(t, config)
+	events, err := f.Fetch()
+	event := events[0]
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), event.StringToPrint())
+
+	assert.EqualValues(t, "exchange.name", event["name"])
+	assert.EqualValues(t, "/", event["vhost"])
+	assert.EqualValues(t, true, event["durable"])
+	assert.EqualValues(t, false, event["auto_delete"])
+	assert.EqualValues(t, false, event["internal"])
+
+	messages := event["messages"].(common.MapStr)
```

Comment:
I wonder if for the comparison here you could create a `common.MapStr` with all fields inside with the expected values and then compare it? Would probably make the code a bit more readable and easier to extend.

Reply:
I tried doing that, but the Equals comparison fails for some reason, although the maps are identical (albeit in a different order): 
```
	Error:      	Not equal:
	            	expected: common.MapStr{"return_unroutable":common.MapStr{"count":40, "details":common.MapStr{"rate":123.123}}, "redeliver":common.MapStr{"count":30, "details":common.MapStr{"rate":0.123}}, "publish":common.MapStr{"count":123, "details":common.MapStr{"rate":0.1}}, "publish_in":common.MapStr{"count":100, "details":common.MapStr{"rate":0.5}}, "publish_out":common.MapStr{"count":99, "details":common.MapStr{"rate":0.9}}, "ack":common.MapStr{"count":60, "details":common.MapStr{"rate":12.5}}, "deliver_get":common.MapStr{"details":common.MapStr{"rate":43.21}, "count":50}, "confirm":common.MapStr{"count":120, "details":common.MapStr{"rate":98.63}}}
	            	actual  : common.MapStr{"ack":common.MapStr{"count":60, "details":common.MapStr{"rate":12.5}}, "deliver_get":common.MapStr{"count":50, "details":common.MapStr{"rate":43.21}}, "confirm":common.MapStr{"count":120, "details":common.MapStr{"rate":98.63}}, "return_unroutable":common.MapStr{"count":40, "details":common.MapStr{"rate":123.123}}, "redeliver":common.MapStr{"count":30, "details":common.MapStr{"rate":0.123}}, "publish":common.MapStr{"count":123, "details":common.MapStr{"rate":0.1}}, "publish_in":common.MapStr{"count":100, "details":common.MapStr{"rate":0.5}}, "publish_out":common.MapStr{"details":common.MapStr{"rate":0.9}, "count":99}}

```